addons/map/functions/fnc_getUnitFlashlights.sqf

```diff
@@ -18,16 +18,17 @@
 params ["_unit"];
 
 private _flashlights = [];
+private _cfgWeapons = configFile >> "CfgWeapons";
 
 {
     private _item = _x;
-    private _weaponConfig = configFile >> "CfgWeapons" >> _item;
+    private _weaponConfig = _cfgWeapons >> _item;
     {
         if (
             isText (_x >> "ACE_Flashlight_Colour")
             || {!(getArray (_x >> "ambient") in [[], [0,0,0]])}
-        ) then {
-            _flashlights pushBackUnique _item;
+        ) exitWith {
```

Comment:
sounds like you want `findIf` here

Reply:
yeah but then decided `exitWith` is more logical here.

Reply:
but less efficient and IMO less clear? Atleast from my perspective

Reply:
I only got you now. Can't say it's way more clear but should be faster.